.github/workflows/cd.yaml

```diff
@@ -115,111 +111,126 @@ jobs:
+        run: Add-Content $env:GITHUB_PATH "${{ github.workspace }}"
+      - name: Set Path
+        if: runner.os != 'Windows'
+        run: echo "$GITHUB_WORKSPACE" >> $GITHUB_PATH
 
       - name: Install Dependencies and Run Tests
+        working-directory: functional-tests
         run: |
-          python -m pip install --force-reinstall -v pip==23.3.1
+          python -m pip install --force-reinstall -v pip==25.3
           pip install -r requirements.txt
           pytest
-        working-directory: packages/cli/functional-tests
 
   github-release:
-    needs: [release, package]
-
-    if: needs.release.outputs.cli-released == 'true'
+    needs: [npm-release, package]
 
     name: Create Github Release
 
     runs-on: ubuntu-latest
 
     steps:
+      - name: Debug Version
+        run: |
+          echo "CLI Version: ${{ needs.npm-release.outputs.cli-version }}"
+          echo "CLI Tag: ${{ needs.npm-release.outputs.cli-tag }}"
+
       - name: Checkout Repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v6.0.1
 
       - name: Download Artifacts
-        uses: actions/download-artifact@v4.1.8
+        uses: actions/download-artifact@v7.0.0
         with:
-          name: dist_bin
+          name: Binaries
 
-      - name: Extract Artifacts
-        run: tar -xvf dist_bin.tar
+      - name: Debug Downloaded Artifacts
+        run: ls -l dist_bin/smartthings-*.{tgz,zip}
 
       - name: Setup Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v6.1.0
         with:
           node-version: 24.8.0
 
-      - run: npm ci
-
-      # hack since Github auto generated notes aren't working right now
-      - name: Generate Github Release Notes
-        run: node generate-release-notes.mjs ${{ needs.release.outputs.cli-version }}
-
       - name: Create Github Release
-        uses: softprops/action-gh-release@v1
+        uses: softprops/action-gh-release@v2.5.0
         with:
-          name: ${{ needs.release.outputs.cli-version }}
-          body_path: ${{ github.workspace }}/RELEASE_NOTES.txt
-          tag_name: ${{ needs.release.outputs.cli-tag }}
+          name: ${{ needs.npm-release.outputs.cli-version }}
+          tag_name: ${{ needs.npm-release.outputs.cli-tag }}
+          generate_release_notes: true
           prerelease: false
-          files: 'packages/cli/dist_bin/**/*.@(tgz|zip)'
+          files: 'dist_bin/smartthings-*.{tgz,zip}'
 
   homebrew-formula:
-    needs: [release, github-release]
-
-    if: needs.release.outputs.cli-released == 'true'
+    needs: [npm-release, github-release]
 
     name: Bump Homebrew Formula
 
     runs-on: macos-latest
 
     steps:
     - name: Checkout Repo
-      uses: actions/checkout@v3
+      uses: actions/checkout@v6.0.1
 
     - name: Configure Git Identity
-      uses: Homebrew/actions/git-user-config@master
+      uses: Homebrew/actions/git-user-config@main
```

Comment:
It's gross that we have to use `main` here rather than a specific version. This action doesn't seem to use versions though. We might want to look for alternatives at some point.